### PR TITLE
[Azure] Add sovereign cloud support for log analytics

### LINKED ISSUE #856: [Azure] Add sovereign cloud support for log analytics
The Base URI is incorrectly resolved for the log analytics client, and should be fixed to read from the environment.

Addresses #811

### PR DESCRIPTION
**Pull request by davidkpiano**
This PR resolves the Base URI for the log analytics module to support sovereign cloud.

Addresses #811, fixes #856

cc. @rguthriemsft

### CODE REVIEW

**Comment by hattan** [comment]
[Microsoft CI Bot] TL;DR; success :thumbsup:

You can check the status of the CI Pipeline logs here ; https://github.com/aztfmod/terratest/actions/runs/730443366

File: modules/azure/client_factory.go [hunk 1]
```diff
@@ -206,6 +208,31 @@ func CreateStorageBlobContainerClientE(subscriptionID string) (*storage.BlobCont
 	return &blobContainerClient, nil
 }
 
+=======
```
Comment (HadwaAbdelhalem) [inline modules/azure/client_factory.go:211]: @davidkpiano , please cleanup the merge conflicts comments

File: modules/azure/client_factory.go [hunk 2]
```diff
@@ -160,6 +161,7 @@ func CreateKeyVaultManagementClientE(subscriptionID string) (*kvmng.VaultsClient
 	return &vaultClient, nil
 }
```
Comment (HadwaAbdelhalem) [inline modules/azure/client_factory.go:163]: please cleanup the merge conflicts comments

**Review by HadwaAbdelhalem — CHANGES_REQUESTED** [review]
Thanks @davidkpiano , Mostly LGTM please cleanup the merge conflicts comments

**Comment by davidkpiano** [comment]
Superseded by #884

**Pull request closed without merging**